.github/workflows/Actions-workflow.yml

```diff
@@ -0,0 +1,241 @@
+                    job['name']=name
+
+                  elif line.find('uses:')!=-1 :
+                    uses=line.split(':')[1]  
+                    uses=uses.strip(' ') 
+                    job['uses']=uses
+
+                  elif line.find('needs:')!=-1 :
+                    needs=line.split(':')[1]
+                    needs=needs.strip(" ")
+                    job['needs']=needs
+
+                  elif line.find('with:')!=-1 :
+                    columnNumOfWith=line.find('with:')
+                    line=getNextLine(fileContent)
+                    while(len(line)>columnNumOfWith and line[columnNumOfWith]==" " ):
+                      input=line.strip(" ")
+                      if(len(input)>30):
+                        input=input[:30]+'...'
+                      job['inputs'].append(input)
+                      line=getNextLine(fileContent)
+
+                  if(line.find('steps:')!=-1 or line=="EOF" or line[columnNum]!=" "):
+                    break
+                  line=getNextLine(fileContent)
+          
+          def getWorkflowTriggers(yamlFile,file):
+            workflowDetails[file]['trigger']=[]
+            fileContent=yamlFile.getFileContent()
+            for line in fileContent:
+              if line.strip("\n ").startswith("on:"):
+                line=line.split(':')[1]
+                line=line.strip("\n ")
+                if line and not line.startswith('#'):       # This case handles triggers which are declared on the same line of "on:" key example: on: [push, pull_request] 
+                  line=line.replace("[","")
+                  line=line.replace("]","")
+                  line=line.replace(" ","")
+                  line=line.replace('#',',#')
+                  for trigger in line.split(','):
+                    if not trigger.startswith('#'):
+                      workflowDetails[file]['trigger'].append(trigger)
+                else :                                     # Here it handles triggers that are declared on the next line of "on:" key 
+                  columnNumOfOn=0       
+                  line=getNextLine(fileContent)
+                  subColumnNum=getColumnNum(line)
+                  while(line[columnNumOfOn]==" "):
+                    if(line[subColumnNum]!=" "):
+                      trigger=line.strip(" ")
+                      trigger=trigger.split(':')[0]
+                      workflowDetails[file]['trigger'].append(trigger)
+                    line=getNextLine(fileContent)
+        
+          workflowsDir='.github/workflows/'
+          yaml_files=glob.glob(workflowsDir+'*.y*ml')
+          workflowDetails={}  #This is the main dictionary where all the workflow details are stored
+          
+          for file in sorted(yaml_files):
+            file=file.split('/')[-1]
+            getWorkflowName(workflowsDir,file)
+          
+          markdown=workflowsDir+'ActionFlow.md'
+
+          with open( markdown ,'a') as markdownFile:
```

Comment:
Why do you open the markdown file for append? With that mode every time the markdown file might grow and hold old diagrams as well.

Reply:
On line 30, I will clear the file. So, all the previous graphs will be erased.

Reply:
Yes, you are right.